File: benchmarks/c/03_matrix_mul.c
// Matrix Multiplication Benchmark - C Implementation
// Multiply two NxN matrices (matching Viper algorithm)

#include <stdio.h>

int main() {
    int n = 30;
    long checksum = 0;
    
    for (int row = 0; row < n; row++) {
        for (int col = 0; col < n; col++) {
            int sum = 0;
            for (int k = 0; k < n; k++) {
                int a_val = (row * n + k) % 10;
                int b_val = (k * n + col + 1) % 10;
                sum += a_val * b_val;
            }
            checksum += sum;
        }
    }
    
    printf("matrix %dx%d checksum: %ld\n", n, n, checksum);
    return 0;
}

File: benchmarks/go/03_matrix_mul.go
// Matrix Multiplication Benchmark - Go Implementation
// Multiply two NxN matrices (matching Viper algorithm)

package main

import "fmt"

func main() {
	n := 30
	checksum := 0
	
	for row := 0; row < n; row++ {
		for col := 0; col < n; col++ {
			sum := 0
			for k := 0; k < n; k++ {
				a_val := (row*n + k) % 10
				b_val := (k*n + col + 1) % 10
				sum += a_val * b_val
			}
			checksum += sum
		}
	}
	
	fmt.Printf("matrix %dx%d checksum: %d\n", n, n, checksum)
}
